Beware custom `fail_calc` in tests that have the potential to return no rows at all (#5246)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/fail_calc.md

For instance, you can configure a `unique` test to return `sum(n_records)` instead of `count(*)` as the failure calculation: that is, the number of rows in the model containing a duplicated column value, rather than the number of distinct column values that are duplicated.

:::tip Tip
Beware using functions like `sum()` for `fail_calc` in any test that has the potential to return no rows at all.

> Such a test would break (neither pass nor fail), and return the following error: 

Reviewer: ```suggestion
If no rows are returned, the test won't pass or fail but will return the following error: 
```

```
None is not of type 'integer'

Failed validating 'type' in schema['properties']['failures']:
    {'type': 'integer'}

On instance['failures']:
    None
```

> This scenario can be handled with a case statement that returns `0` when there are no rows:

Reviewer: ```suggestion
To avoid this issue, use a case statement to ensure that `0` is returned when no rows exist:
```
